cursor_resolver.go

```diff
@@ -10,6 +10,8 @@ import (
 	"go.uber.org/zap"
 )
 
+var _ Stepable = &wrappedObject{}
```

Comment:
Usually for pointer stuff, the following construct is used:

```suggestion
var _ Stepable = (*wrappedObject)(nil)
```

Nitpicking